tls/s2n_handshake_io.c

```diff
@@ -1232,14 +1232,24 @@ static int s2n_handshake_read_io(struct s2n_connection *conn)
 
         /* Call the relevant handler */
         r = ACTIVE_STATE(conn).handler[conn->mode] (conn);
+        /* At this point we may have already failed.
```

Comment:
I think this paragraph can be re-worded:
"If the handler fails, we clean up the handshake and skip processing steps necessary to continue connecting (such as updating the transcript hash.)"